src/foam/nanos/crunch/Capability.js

```diff
@@ -307,6 +311,73 @@ foam.CLASS({
 
       return today.after(capabilityExpiry);
       `
+    },
+    {
+      name: 'getJunction',
+      type: 'foam.nanos.crunch.UserCapabilityJunction',
+      documentation: `
+        answers the question "does the context have this capability granted?"
+      `,
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      javaCode: `
+        User user = ((Subject) x.get("subject")).getUser();
+        User realUser = ((Subject) x.get("subject")).getRealUser();
+
+        Predicate acjPredicate = INSTANCE_OF(AgentCapabilityJunction.class);
+        Predicate targetPredicate = EQ(UserCapabilityJunction.TARGET_ID, getId());
+        try {
+          DAO capabilityDAO = ( x.get("localCapabilityDAO") == null ) ? (DAO) x.get("capabilityDAO") : (DAO) x.get("localCapabilityDAO");
```

Comment:
could even do this to simplify the code a bit - by removing targetPredicate
```suggestion
          DAO capabilityDAO = (DAO) x.get("localCapabilityDAO").where(EQ(UserCapabilityJunction.TARGET_ID, getId()));
```

Reply:
targetPredicate is re-used